lib/gollum-lib/filter/sourcemap.rb

```diff
@@ -0,0 +1,24 @@
+# ~*~ encoding: utf-8 ~*~
+
+# Source Maps
+#
+# Find lines corresponding to h1 headers for the present markup and insert markers indicating the line number in the raw source.
+
+class Gollum::Filter::SourceMap < Gollum::Filter
+  def extract(data)
+  	regex = Gollum::Markup.formats[@markup.format][:h1]
+    return data if regex.nil?
+    i = 0
+    data.lines.map do |line|
```

Comment:
Indeed. I don't think we can use MRI to directly "one-line" a solution that maps sections to a source line. On the other hand, it may not even be the best solution - see below for the note about offsets.